lib/livebook/zta/livebook_teams.ex

```diff
@@ -111,6 +113,27 @@ defmodule Livebook.ZTA.LivebookTeams do
     end
   end
 
+  defp authorize_user({%{halted: true} = conn, metadata}, _team) do
+    {conn, metadata}
+  end
+
+  defp authorize_user({%{path_info: path_info} = conn, metadata}, _team)
+       when path_info in [[], ["apps"]] do
+    {conn, metadata}
+  end
```

Comment:
I'm not sure if this should be a part of the ZTA implementation. The purpose behind ZTA plug is very focused.

For apps we have `AppAuthHook`, so we could have the extra authorization there. For the full access, we may need a separate plug/hook.

@josevalim wdyt?

Reply:
The full access should be checked by ZTA (as that is generic). The app access should be done in a generic way, where we pass the user metadata (groups) and the app authorization groups to Livebook.Apps and be outside of this plug.

Reply:
@josevalim if someone doesn't have full access, we still need to pass them through here in case they are accessing apps paths. But at the same time, my point is that we should not be special casing app paths here, because we need to be careful to keep them in sync with the router.

Reply:
Exactly. The best way to think about it is that ZTA does _Authentication_ and, in this case, the authentication applies to the whole deployment group, so we perform the full access check here.

The app one we can do in AppAuthHook and also in the app listing.

PS: Note that we do user_plug and then auth_plug. But for hooks we do AuthHook and UserHook. We may need to swap the order of the hooks.

Reply:
*I deleted my previous comments to avoid confusion*.

Yes, you are correct. Here is my suggestion, rename user.groups to user.restricted_apps_groups. Then I think the logic here should be this:

1. If the user has full access, pass the user forward with user.restricted_apps_groups == nil
2. If the user does not have full access, set `user.restricted_apps_groups` to the groups
3. In AuthPlug check if `user.restricted_apps_groups != nil`, if it is, they can only access apps, and therefore we should fail
4. In AppAuthHook check if `user.restricted_apps_groups != nil`, if so, it only access if the user belongs to the app groups

In other words, the goal of the ZTA is to populate `restrict_apps_groups`accordingly, the other plugs redirect.

Reply:
Sounds good!